src/docfx/template/TemplateEngine.cs

```diff
@@ -5,34 +5,37 @@
 using System.Collections.Concurrent;
 using System.Collections.Generic;
 using System.IO;
-using System.Threading;
 using System.Threading.Tasks;
 using Newtonsoft.Json.Linq;
 
 namespace Microsoft.Docs.Build
 {
     internal class TemplateEngine
     {
+        public TemplateEngineRunner TemplateEngineRunner { get; init; }
+
```

Comment:
If we are to split `TemplateEngine`, consider split it into a `JsonSchemaStore` and a `TemplateEngine`, where `JsonSchemaStore` manages all the schemas, and `TemplateEngine` runs `liquid/mustache/js`.

Consider a separate PR just to do the split.